compiler/test/compilable/test22760.d

```diff
@@ -0,0 +1,16 @@
+// https://issues.dlang.org/show_bug.cgi?id=22760
+
+extern(C++) void f(T)(T)
+{
+}
+struct S1(T)
+{
+    struct S2
+    {
+    }
+}
+int main()
```

Comment:
try to avoid calling function `main` in compilable test, otherwise looks fine.

Reply:
It makes linking slower

Reply:
IIRC compilable tests are not linked, but I might be wrong

Reply:
the problem is that its misleading, not that there is anything technically wrong with it.

Reply:
ok, done.